### ISSUE TITLE
[BUG] Snackbar InvalidOperationException when using dotnet trimmer

### ISSUE THREAD

**Issue by tranb3r** [CommunityToolkit/Maui#1460]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

When attempting to show a `Snackbar` on a Maui app for Android with dotnet trimmer `full` mode enabled, the following exception occurs:
```
System.InvalidOperationException: Unable to find Snackbar text view
  at CommunityToolkit.Maui.Alerts.Snackbar.SetMessageForView(View&, IFontManager)
  at CommunityToolkit.Maui.Alerts.Snackbar.ShowPlatform(CancellationToken )
  at MauiAppSnackbar.MainPage.OnCounterClicked(Object sender, EventArgs e)
```

### Expected Behavior

Instead of throwing an exception, the `Snackbar` should be displayed.

Workaround: do not trim CommunityToolkit.Maui
```
<ItemGroup Condition="'$(Configuration)|$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))'=='Release|android'">
	<TrimmerRootAssembly Include="CommunityToolkit.Maui" RootMode="All" />
</ItemGroup>
```

### Steps To Reproduce

1. Open the repro project
2. Launch in release mode (dotnet trimmer full mode)
3. Click on button: a dialog will appear, catching the exception that is raised by CommunityToolkit.Maui

### Link to public reproduction project repository

https://github.com/tranb3r/Issues/tree/main/MauiAppSnackbar

### Environment

```markdown
- .NET MAUI CommunityToolkit: 6.0.0
- OS: Android
- .NET MAUI: 8.0 rc2
```

**Comment by pictos** [comment]
@tranb3r The MCT isn't linker friendly, so when you turn the linker on, you should take care on what Linker shouldn't remove. You can create a xml file with the methods and assemblies that you want to keep, as you can see [here](https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options?pivots=dotnet-7-0#root-descriptors)

**Comment by tranb3r** [comment]
@pictos 
MCT is now pretty much the only third-party lib I'm using that is still not linker friendly. Is there a plan to improve this?
For the specific issue of snack-bar, right now as a workaround I'm rooting the whole MCT assembly. This is fine, but maybe I could set a more specific trimmer directive?

**Comment by tranb3r** [comment]
Also, if the MCT is not trimmer ready, maybe it should simply have its IsTrimmable attribute set. This would solve this issue.

**Comment by brminnick** [comment]
>  if the MCT is not trimmer ready, maybe it should simply have its IsTrimmable attribute set

@VladislavAntonyuk - could you add `<IsTrimmable>false</IsTrimmable>`  to  #1367?

```xml
<PropertyGroup>
    <IsTrimmable>false</IsTrimmable>
</PropertyGroup>
```

Here's the .NET 8 docs for it:
https://learn.microsoft.com/dotnet/core/deploying/trimming/prepare-libraries-for-trimming?pivots=dotnet-8-0#enable-project-specific-trimming

**Comment by brminnick** [comment]
I don't disagree, Pedro. 

One data point: the size of the NuGet Package for `CommunityToolkit.Maui` is 1.15MB and the `CommunityToolkit.Maui.Core` is 697.48 KB. To me this size is negligible without Trimming.

That being said, if anyone wants to open a PR to enable Trimming, I'd be more than happy to review + merge it!

**Comment by tranb3r** [comment]
@pictos 
Let me clarify my question. 
Right now, I workaround this issue by preserving the whole MCT assembly. 
However it's quite big. So I'd like to set a more specific directive, targeting this snackbar exception.
But so far I haven't been able to figure out which class/method/... should be added to the trimmer descriptors file.
Any idea?

**Comment by tranb3r** [comment]
By the way, the descriptors doc is here: https://github.com/dotnet/linker/blob/main/docs/data-formats.md

**Comment by pictos** [comment]
> @pictos Let me clarify my question. Right now, I workaround this issue by preserving the whole MCT assembly. However it's quite big. So I'd like to set a more specific directive, targeting this snackbar exception. But so far I haven't been able to figure out which class/method/... should be added to the trimmer descriptors file. Any idea?

@tranb3r this is more a cat/ mouse game... Like the error will give you a hint about what is missing, in this case, is the ` Snackbar text view` (from your report), then you need to dig into the code and see where this is located and preserve it on nuget... Another thing is to build in release and inspect/decompile the `CommunityToolkit.Maui.dll` and see what's missing and including. Since XF I didn't know any other (easy) way to do it. Grab a drink and hope you find this challenge fun

**Comment by tranb3r** [comment]
@pictos 
I know the game, I've been doing it for many years.
However, for this specific exception, I've not managed to find out what is missing. So I was hoping someone could help me.
I understand you do not have the time for that. Maybe we can keep this issue open until someone can find the solution?

**Comment by tranb3r** [comment]
@pictos 
I've investigated this issue a bit more.
There is no issue when building for net7. This is a regression for net8.
I think the issue is related to the new resource designer assembly. This would explain why no matter how I try to "preserve" MCT code, the issue is still here, unless I root the whole assembly.
What do you think? Maybe this issue could be transfered to the xamarin-android team?

**Comment by pictos** [comment]
@tranb3r thanks for looking into it, we don't support .net8 bits yet, so not sure if could be that... Maybe, when we set our TFM to net8 the issue goes away, but I don't know.

Hey @jonathanpeppers, do you know something that may changed on Linker on the latest android bits?

**Comment by jonathanpeppers** [comment]
It could be related to this, which will be fixed in .NET 8 GA: https://github.com/xamarin/xamarin-android/commit/03018e02d0e33eb37309ae83fd9c6ad549fe1ce5

There might be an intermediate release before GA, where we have a chance to ship this change.

**Comment by tranb3r** [comment]
> It could be related to this, which will be fixed in .NET 8 GA: [xamarin/xamarin-android@03018e0](https://github.com/xamarin/xamarin-android/commit/03018e02d0e33eb37309ae83fd9c6ad549fe1ce5)
> There might be an intermediate release before GA, where we have a chance to ship this change.

@jonathanpeppers 
I've built my app with the new version of xamarin-android (34.0.0-rc.2.479) and the issue is still here.

@brminnick 
Why did you close this issue? How is this bug related to the .NET8 migration of the toolkit?

**Comment by brminnick** [comment]
> @brminnick
> Why did you close this issue? How is this bug related to the .NET8 migration of the toolkit?

In https://github.com/CommunityToolkit/Maui/pull/1367, we added `<IsTrimmable>false</IsTrimmable>` making the .NET MAUI Community Toolkit safe to use with the Trimmer.

We discussed this in this Issue above:  https://github.com/CommunityToolkit/Maui/issues/1460#issuecomment-1765193944

Next time, please review the PR code and the comment history in the Issue before tagging me. We are all volunteers working on the CommunityToolkit; we don't get paid by Microsoft to do this work, and our time is limited. 

https://github.com/CommunityToolkit/Maui/blob/fa59686347670b906ec79e6cfa84d0782c64fb7b/Directory.Build.props#L14

**Comment by tranb3r** [comment]
Sorry for tagging you but as the issue wa closed it was probably the only way to get your attention.
As this issue and the fix are not at all related to net8, I honestly thought it deserved some kind of explanation. Maybe next time it would be better to do a separate PR in a similar situation.
Also, I regret that the issue was closed without a proper investigation. This is probably hiding a bug in xamarin-android, as JP was suspecting. So I was sincerly hoping that someone could come up with a better fix. Even if IsTrimmable=false is fine for now.
Finally, I'm a long time xamarin user, and I have to say that MCT is great. I'm really thankful for your time and your work.

**Comment by jonathanpeppers** [comment]
Note that setting `IsTrimmable=false` is not a long-term solution. It will not work on NativeAOT for iOS, for example, as NativeAOT *has* to trim everything. It will completely ignore this setting in this case.

Details at: https://github.com/dotnet/maui/issues/18658

**Comment by brminnick** [comment]
Noted- thanks @jonathanpeppers!

I’ll open an Issue and we’ll certainly be sure to add Trimmer support before iOS moves to NativeAOT in .NET 9 💯

**Comment by tranb3r** [comment]
This issue is fixed in version 7.0.0 of CommunityToolkit.Maui.
Probably thanks to the bug fix in xamarin-android mentionned by Jonathan.

Please note that the added `IsTrimmable=false` does not seem to have any effect. The assembly-level attribute is not set.
This is totally fine in the context of this issue: the bug is fixed and we can still trim the assembly.
